hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -341,15 +341,14 @@ public String getIdentityColumn() {
 		}
 
 		final String[] propertyNames = getIdentifierPropertyNames();
-		List<String> columns = new ArrayList<>();
+		ArrayList<String> columns = new ArrayList<>( propertyNames.length );
 		final boolean inSelect = getWalker().getStatementType() == HqlSqlTokenTypes.SELECT;
 		for ( String propertyName : propertyNames ) {
 			String[] propertyNameColumns = toColumns( table, propertyName, inSelect );
-			for ( String propertyNameColumn : propertyNameColumns ) {
-				columns.add( propertyNameColumn );
-			}
+			columns.ensureCapacity( columns.size() + propertyNameColumns.length );
+			Collections.addAll( columns,propertyNameColumns );
 		}
-		return columns.toArray( new String[columns.size()] );
+		return columns.toArray( new String[0] );
```

Comment:
Interesting. see https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size

Reply:
Yes, in recent java versions using the zero sized constructor is the fastest way.

Reply:
it seems after you fix the style issue your PR has good chance tobe merged. why not continue perfecting the PR to have it finalized?

Reply:
it seems after you fix the style issue your PR has good chance tobe merged. why not continue perfecting the PR to have it finalized?